numpy/lib/_npyio_impl.py

```diff
@@ -1559,84 +1559,101 @@ def first_write(self, v):
     if isinstance(fname, os.PathLike):
         fname = os.fspath(fname)
     if _is_string_like(fname):
-        # datasource doesn't support creating a new file ...
+        # Ensure the file exists before opening it with np.lib._datasource.
         open(fname, 'wt').close()
         fh = np.lib._datasource.open(fname, 'wt', encoding=encoding)
         own_fh = True
     elif hasattr(fname, 'write'):
-        # wrap to handle byte output streams
+        # Wrap to handle byte output streams
         fh = WriteWrap(fname, encoding or 'latin1')
     else:
-        raise ValueError('fname must be a string or file handle')
+        raise ValueError("fname must be a string or file handle")
 
     try:
         X = np.asarray(X)
-
-        # Handle 1-dimensional arrays
         if X.ndim == 0 or X.ndim > 2:
-            raise ValueError(
-                "Expected 1D or 2D array, got %dD array instead" % X.ndim)
+            raise ValueError(f"Expected 1D or 2D array, got {X.ndim}D array instead")
         elif X.ndim == 1:
-            # Common case -- 1d array of numbers
             if X.dtype.names is None:
                 X = np.atleast_2d(X).T
                 ncol = 1
-
-            # Complex dtype -- each field indicates a separate column
             else:
                 ncol = len(X.dtype.names)
         else:
             ncol = X.shape[1]
 
         iscomplex_X = np.iscomplexobj(X)
-        # `fmt` can be a string with multiple insertion points or a
-        # list of formats.  E.g. '%10.5f\t%10d' or ('%10.5f', '$10d')
-        if type(fmt) in (list, tuple):
-            if len(fmt) != ncol:
-                raise AttributeError('fmt has wrong shape.  %s' % str(fmt))
-            format = delimiter.join(fmt)
-        elif isinstance(fmt, str):
-            n_fmt_chars = fmt.count('%')
-            error = ValueError('fmt has wrong number of %% formats:  %s' % fmt)
-            if n_fmt_chars == 1:
-                if iscomplex_X:
-                    fmt = [' (%s+%sj)' % (fmt, fmt), ] * ncol
-                else:
-                    fmt = [fmt, ] * ncol
-                format = delimiter.join(fmt)
-            elif iscomplex_X and n_fmt_chars != (2 * ncol):
-                raise error
-            elif ((not iscomplex_X) and n_fmt_chars != ncol):
-                raise error
+
+        if fstring_fmt:
+            if isinstance(fstring_fmt, str):
+                if "{" in fstring_fmt and "}" in fstring_fmt:
+                    if fstring_fmt.count("{") > 1 or fstring_fmt.count("}") > 1:
+                        raise ValueError(
+                            "Multi-column f-strings are not supported for fstring_fmt. "
+                            "Please provide a single f-string for formatting."
+                        )
```

Comment:
What about `fstring_fmt = '{{{:.3f}}}'`? This is a valid (single-column) f-string format, but rejected here. I would maybe just skip all the checks. If the format is incorrect, an error will be generated later.